am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -222,6 +225,18 @@ private boolean explicitAccessType(Set<String> userRoles) {
         return roleBasedAccessRecords.stream().collect(getMapCollector());
     }
 
+    /**
+     * Retrieves a list of {@link ResourceDefinition} that user roles have create permissions for.
+     *
+     * @param userRoles a set of roles
+     * @return a list of resource definitions
+     */
+    public List<ResourceDefinition> getResourceDefinitionsWithCreatePermission(Set<String> userRoles) {
+        return jdbi.withExtension(AccessManagementRepository.class, dao ->
+            userRoles.stream().map(dao::resourceCreationAllowedList).flatMap(Collection::stream)
```

Comment:
I suggest having `map` and `collect` on separate lines that IMO helps understanding transformations.

Wouldn't it be better to pass set of roles to JDBI and having one query instead of n (where n is number of roles).

Does `.toList()` protect us from duplicates e.g. in scenario where role A and role B return same resource definition. If we had one query distinct would nicely dedicate on DB level without needed to read and deserialise that much.